src/main_nep/nep3.cu

```diff
@@ -308,7 +308,25 @@ void NEP3::update_potential(float* parameters, ANN& ann)
     pointer += ann.num_neurons1;
   }
   ann.b1 = pointer;
-  ann.c = ann.b1 + 1;
+  pointer += 1;
+
+  if (para.train_mode == 2) {
+    for (int t = 0; t < paramb.num_types; ++t) {
+      if (t > 0 && paramb.version != 4) { // Use the same set of NN parameters for NEP2 and NEP3
+        pointer -= (ann.dim + 2) * ann.num_neurons1;
+      }
+      ann.w0_pol[t] = pointer;
+      pointer += ann.num_neurons1 * ann.dim;
+      ann.b0_pol[t] = pointer;
+      pointer += ann.num_neurons1;
+      ann.w1_pol[t] = pointer;
+      pointer += ann.num_neurons1;
+    }
+    ann.b1_pol = pointer;
+    pointer += 1;
+  }
+
+  ann.c = pointer;
```

Comment:
Is there a written description of the polarizability model? Is it the same neural network architecture as in regular NEP, or is it modified in order to be equivariant? I'm curious on how it's implemented :)

Reply:
This is still kind of under development (a paper is being prepared) and will be documented in GPUMD-3.7 I think. 

Basically, it only involves very minor modifications of the NEP model of potential energy, to get a model for polarizability as well as dipole.

Let's discuss this via email for more technical details.